src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -0,0 +1,762 @@
+
+    deltaMousePosition.copy(mousePosition).sub(lastMousePosition);
+
+    lastMousePosition.copy(mousePosition);
+
+    if (state === STATE.ROTATE)
+    { this.handleRotation(); }
+    else if (state === STATE.DRAG)
+    { this.handleDragMovement(); }
+    else if (state === STATE.PAN)
+    { this.handlePanMovement(); }
+};
+
+/** =========================================================================================================================
+* Catch and manage the event when a key is up.
+* @param {event} event : the current event
+*/
+var onKeyUp = function onKeyUp(event) {
+    if (event.keyCode === keys.CTRL) {
+        isCtrlDown = false;
+    }
+};
+
+/** =========================================================================================================================
+* Catch and manage the event when a key is down.
+* @param {event} event : the current event
+*/
+var onKeyDown = function onKeyDown(event) {
+    if (state === STATE.TRAVEL) {
+        return;
+    }
+    if (event.keyCode === keys.T) {
+        this.goToTopView();
+    }
+    if (event.keyCode === keys.S) {
+        this.goToStartView();
+    }
+    if (event.keyCode === keys.SPACE) {
+        this.initiateSmartZoom(event);
+    }
+    if (event.keyCode === keys.CTRL) {
+        isCtrlDown = true;
+    }
+};
+
+/** =========================================================================================================================
+* Catch and manage the event when the mouse wheel is rolled.
+* @param {event} event : the current event
+*/
+var onMouseWheel = function onMouseWheel(event) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (state === STATE.NONE) {
+        this.initiateZoom(event);
+    }
+};
+
+/** =========================================================================================================================
+* Catch and manage the event when the context menu is called (by a right click on the window).
+* We use this to prevent the context menu from appearing, so we can use right click for other inputs.
+* @param {event} event : the current event
+*/
+var onContextMenu = function onContextMenu(event) {
+    event.preventDefault();
+};
+
+
+/** =========================================================================================================================
+* smoothing function (sigmoid) : based on h01 Hermite function
+* returns a value between 0 and 1
+* @param {number} value : the value to be smoothed, between 0 and 1
+* @returns {number}
+*/
+var smooth = function smooth(value) {
+    // p between 1.0 and 1.5
+    return Math.pow((value * value * (3 - 2 * value)), 1.20);
+};
+
+/** =========================================================================================================================
+* clamp value between 0 and 1
+* @param {number} value : the value to be smoothed, between 0 and 1
+* @returns {number}
+*/
+var clamp01 = function clamp01(value) {
```

Comment:
You should directly use `THREE.Math.clamp(value, 0, 1)` instead of adding a one liner function.